packages/commerce-server/src/format-prices-for-product.ts

```diff
@@ -31,6 +33,60 @@ type FormatPricesForProductOptions = {
   userId?: string
 }
 
+async function getFixedDiscountForUpgrade({
+  upgradeFromPurchaseId,
+  ctx = defaultContext,
+}: {
+  upgradeFromPurchaseId: string
+  ctx: Context
+}) {
+  let unitPrice = 0
+  const {getPurchase} = getSdk({ctx})
+
+  const {getCheckoutSession} = getStripeSdk()
+  const upgradeFromPurchase = await getPurchase({
+    where: {
+      id: upgradeFromPurchaseId,
+    },
+    select: {
+      id: true,
+      bulkCoupon: {
+        select: {
+          id: true,
+        },
+      },
+      redeemedBulkCouponId: true,
+      totalAmount: true,
+      productId: true,
+    },
+  })
+
+  if (upgradeFromPurchase && upgradeFromPurchase?.redeemedBulkCouponId) {
+    const parentPurchase: (Purchase & {merchantSession: any}) | null =
+      (await getPurchase({
+        where: {
+          bulkCouponId: upgradeFromPurchase.redeemedBulkCouponId,
+        },
+        include: {
+          merchantCharge: true,
+          merchantSession: true,
+        },
+      })) as any
+    if (parentPurchase?.merchantSession?.identifier) {
+      const charge = await getCheckoutSession(
+        parentPurchase.merchantSession.identifier,
+      )
+      if (charge.line_items) {
+        const {quantity, amount_total} = charge.line_items.data[0]
+        unitPrice = quantity ? amount_total / 100 / quantity : 0
```

Comment:
Checking my understanding -- the way the _true_ branch of this ternary is working is that the `amount_total` might be something like `236000` and the `quantity = 4`, so then you do `236000 / 100 / 4 => 590` which tells you that each seat cost `590` (hence _unit price_).

And, then for a `total_amount` that involved a bulk discount, you'd have something like `531,000` (which comes from `(590 * 10 * 0.9) * 100`). So then the unit price you'd get here would be the bulk discounted unit price of `531`.